### ISSUE TITLE
Error "cannot create directory" in snap semaphore trying to start

### ISSUE THREAD

**Issue by morganchester** [ansible-semaphore/semaphore#834]
Getting the error trying to start snap semaphore in 2.8.33 and 2.8.35.

Jan 27 08:14:52 * systemd[1]: Started Service for snap application semaphore.semaphored.
Jan 27 08:14:53 * semaphore.semaphored[17025]: mkdir: cannot create directory '/root/snap/semaphore': Not a directory
Jan 27 08:14:53 * semaphore.semaphored[17025]: "host": "/root/snap/semaphore/common/database.boltdb"
Jan 27 08:14:53 * semaphore.semaphored[17025]: "tmp_path": "/root/snap/semaphore/common/repositories",
Jan 27 08:14:53 * systemd[1]: snap.semaphore.semaphored.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jan 27 08:14:53 * systemd[1]: snap.semaphore.semaphored.service: Failed with result 'exit-code'.

Directory exists and everything was fine before

CentOS Linux release 8.5.2111
Snap 2.54.2

Suddenly failed to start.

**Comment by fiftin** [comment]
Hm, did you try install anything else with using Snap?

**Comment by morganchester** [comment]
> Hm, did you try install anything else with using Snap?

Didn't need anything else on snap

Semaphore worked several weeks before it failed to start today. Think it's 2.8.33 that failed. Previous version was 2.8.9 and it was fine

**Comment by fiftin** [comment]
It is looks like Snap issue. Semaphore doesn't create /root/snap/semaphore directory. It's do Snap.

**Comment by fiftin** [comment]
I made fresh install of Semaphore on CentOS using Snap. It's works. I will try reproduce your scenario.

**Comment by fiftin** [comment]
Can't reproduce.

I need your help. 

I see only one line which can lead to this error: `mkdir -p ${SNAP_USER_COMMON}/repositories`.

Looks like, for unknown reason, this command returns error.

**Comment by fiftin** [comment]
Semaphore directory should looks like:
```
[root@test current]# cd /root/snap/semaphore/
[root@test semaphore]# ls
146  150  common  current
```

**Comment by morganchester** [comment]
Yes, it looks just like this. 
The dir /root/snap/semaphore/common/repositories also exists and it's not empty (I'm active semaphore user)

**Comment by fiftin** [comment]
Ok,

1) Does restarting the service help? 
 `sudo snap restart semaphore`

2) Please try execute command?
`
sudo semaphore user list
`
This command uses same directory structure.

**Comment by morganchester** [comment]
1. No, restarting not helps
2. user list response:
````
semaphore user list
Checking db migrations
Executing migration v2.8.26 (at 2022-01-28 07:58:55.617321241 +0200 EET m=+0.123515322)...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x86c6e6]

goroutine 1 [running]:
go.etcd.io/bbolt.(*Bucket).ForEach(0x0, 0xc000143680, 0x1e, 0x20)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/go.etcd.io/bbolt/bucket.go:383 +0x26
github.com/ansible-semaphore/semaphore/db/bolt/migrations.Migration_2_8_28.getProjectRepositories.func1(0xc00012dce0, 0xc000143600, 0xc00012dce0)
	/go/src/github.com/ansible-semaphore/semaphore/db/bolt/migrations/v2.8.26.go:17 +0x107
go.etcd.io/bbolt.(*DB).View(0xc00012a000, 0xc000143730, 0x0, 0x0)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/go.etcd.io/bbolt/db.go:719 +0x93
github.com/ansible-semaphore/semaphore/db/bolt/migrations.Migration_2_8_28.getProjectRepositories(0xc00012a000, 0xc00002af80, 0xa, 0x0, 0xc000143818, 0x4eee0e)
	/go/src/github.com/ansible-semaphore/semaphore/db/bolt/migrations/v2.8.26.go:15 +0x79
github.com/ansible-semaphore/semaphore/db/bolt/migrations.Migration_2_8_28.Apply(0xc00012a000, 0x300, 0x5b)
	/go/src/github.com/ansible-semaphore/semaphore/db/bolt/migrations/v2.8.26.go:59 +0x185
github.com/ansible-semaphore/semaphore/db/bolt.(*BoltDb).ApplyMigration(0xc00000e168, 0xa59409, 0x6, 0x0, 0x0, 0x2, 0x2)
	/go/src/github.com/ansible-semaphore/semaphore/db/bolt/migration.go:40 +0xfe
github.com/ansible-semaphore/semaphore/db.Migrate(0x1890620, 0xc00000e168, 0x0, 0x4)
	/go/src/github.com/ansible-semaphore/semaphore/db/Migration.go:72 +0x3d7
github.com/ansible-semaphore/semaphore/cli/cmd.createStore(0x0, 0x0)
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/root.go:112 +0x134
github.com/ansible-semaphore/semaphore/cli/cmd.glob..func12(0x1b7ff60, 0x1bb9f78, 0x0, 0x0)
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/user_list.go:17 +0x47
github.com/spf13/cobra.(*Command).execute(0x1b7ff60, 0x1bb9f78, 0x0, 0x0, 0x1b7ff60, 0x1bb9f78)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:860 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x1b7f7e0, 0x1b88e50, 0xa59b0b, 0x6)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/ansible-semaphore/semaphore/vendor/github.com/spf13/cobra/command.go:902
github.com/ansible-semaphore/semaphore/cli/cmd.Execute()
	/go/src/github.com/ansible-semaphore/semaphore/cli/cmd/root.go:37 +0x8d
main.main()
	/go/src/github.com/ansible-semaphore/semaphore/cli/main.go:8 +0x25
````

**Comment by fiftin** [comment]
Fixed in 3.8.36. Available in Snap.

**Comment by morganchester** [comment]
Got it working! Thanks!
